integration/spark2/src/main/scala/org/apache/spark/sql/events/MergeBloomIndexEventListener.scala

```diff
@@ -24,59 +24,96 @@ import org.apache.spark.internal.Logging
 import org.apache.spark.sql.SparkSession
 
 import org.apache.carbondata.common.logging.LogServiceFactory
-import org.apache.carbondata.core.datamap.DataMapStoreManager
+import org.apache.carbondata.core.datamap.{DataMapStoreManager, TableDataMap}
 import org.apache.carbondata.core.metadata.schema.datamap.DataMapClassProvider
 import org.apache.carbondata.core.metadata.schema.table.CarbonTable
 import org.apache.carbondata.datamap.CarbonMergeBloomIndexFilesRDD
 import org.apache.carbondata.events._
+import org.apache.carbondata.processing.loading.events.LoadEvents.LoadTablePreStatusUpdateEvent
+import org.apache.carbondata.processing.merger.CarbonDataMergerUtil
 
 class MergeBloomIndexEventListener extends OperationEventListener with Logging {
   val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
 
   override def onEvent(event: Event, operationContext: OperationContext): Unit = {
+    val sparkSession = SparkSession.getActiveSession.get
     event match {
+      case loadPreStatusUpdateEvent: LoadTablePreStatusUpdateEvent =>
+        LOGGER.info("LoadTablePreStatusUpdateEvent called for bloom index merging")
+        // For loading process, segment can not be accessed at this time
+        val loadModel = loadPreStatusUpdateEvent.getCarbonLoadModel
+        val carbonTable = loadModel.getCarbonDataLoadSchema.getCarbonTable
+        val segmentId = loadModel.getSegmentId
+
+        // filter out bloom datamap, skip lazy datamap
+        val bloomDatamaps = DataMapStoreManager.getInstance().getAllDataMap(carbonTable).asScala
+          .filter(_.getDataMapSchema.getProviderName.equalsIgnoreCase(
+            DataMapClassProvider.BLOOMFILTER.getShortName))
+          .filter(!_.getDataMapSchema.isLazy).toList
+
+        mergeBloomIndex(sparkSession, carbonTable, bloomDatamaps, Seq(segmentId))
+
+      case compactPreStatusUpdateEvent: AlterTableCompactionPreStatusUpdateEvent =>
+        LOGGER.info("AlterTableCompactionPreStatusUpdateEvent called for bloom index merging")
+        // For compact process, segment can not be accessed at this time
+        val carbonTable = compactPreStatusUpdateEvent.carbonTable
```

Comment:
seems the following code block is duplicated with line#44~#54, please consider to optimize that.

Reply:
The codes are following similar steps. For L44~L46, it depends on what info we can get from the Event. For L49~L52, do you want to say we can extract a common method get datamap of a table with additional constrains?

Reply:
fine, just keep it as it is